python-backend/app/api/documents/variances/models/variance.py

```diff
@@ -14,6 +15,13 @@ class VarianceDocument(Base):
     variance_id = db.Column(db.Integer,
                             db.ForeignKey('variance.variance_id'),
                             server_default=FetchedValue())
+    uploaded_at = db.Column(db.DateTime, server_default=FetchedValue())
+
+    mine_document = db.relationship('MineDocument', lazy='joined')
+
+    mine_guid = association_proxy('mine_document', 'mine_guid')
+    document_manager_guid = association_proxy('mine_document', 'document_manager_guid')
+    document_name = association_proxy('mine_document', 'document_name')
```

Comment:
haven't seen `assocation_proxy` yet, but this is cool to see, however you are already loading the mine_document relationship, so it's saving you `mine_document.` however that could be very cool to use with the swagger models (show related attributes as being on the object.

Reply:
I'm not totally sure what you're suggesting here. I'm using association proxy here to add those attributes to the variance model so that the marshalling works properly. It references the mine_document relationship

Reply:
Maybe i'm not understanding out assocation_proxy works, i thought it makes an attribute on a related object look like it's on this object. the variance_document swagger model in the mine_api_models doesn't have any of these three. Where are these coming back through marshalling?

Reply:
It does have these. It inherits from the mine document swagger model